### PR TITLE
Prefs [TASK][26310] Handle Event for report screen

### PR DESCRIPTION
**Pull request by toandv98**
## Related Tickets

- [#Task 26310](https://edu-redmine.sun-asterisk.vn/issues/26310)

## WHAT

- ### Handle Event for report screen

## HOW

- Display Report from databse
- Click search button
- RecyclerView item click
- Click new report button

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt [hunk 1]
```diff
@@ -34,4 +36,39 @@ data class Message(
...
+
+    val sendCalendar get() = GregorianCalendar().from(sendTime * 1000)
+
+    val updateCalendar get() = GregorianCalendar().from(updateTime * 1000)
+
+    val isPlan get() = body.contains(PLAN_REGEX.toRegex())
+
+    val isReport get() = body.contains(REPORT_REGEX.toRegex())
+
+    val bodyMessage
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt]:
```suggestion
    val messageBody
```

File: app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt [hunk 2]
```diff
@@ -34,4 +36,39 @@ data class Message(
...
+        get() = body.replace(MESSAGE_TO_REGEX.toRegex(), "")
+            .replace(PLAN_REGEX.toRegex(), "").trim()
+
+    val bodyReport: String
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt]:
```suggestion
    val reportBody: String
```

File: app/src/main/java/com/edu/sun/oereminder/ui/report/ReportAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.edu.sun.oereminder.utils.toDate
+import com.edu.sun.oereminder.utils.toTime
+import kotlinx.android.synthetic.main.item_daily_report.view.*
+
+class ReportAdapter : ListAdapter<Message, ViewHolder>(ReportItemCallback()) {
+
+    private var onItemClick: ((Message) -> Unit)? = null
+    private var unfilteredList = listOf<Message>()
+
+    fun setItemClickListener(listener: (Message) -> Unit) {
```
Comment (tuanpa-0908) [inline app/src/main/java/com/edu/sun/oereminder/ui/report/ReportAdapter.kt]: fun đưa xuống dưới override fun nha

File: app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt [hunk 4]
```diff
@@ -34,4 +36,39 @@ data class Message(
...
+            .replace(PLAN_REGEX.toRegex(), "").trim()
+
+    val messageReport: String
+        get() = PLAN_REGEX.toRegex().find(body)?.value
+            ?.replace(INFO_NOTATION.toRegex(), "")
+            ?.replace(FIRST_LAST_END_LINE.toRegex(), "")
+            ?.replace(END_LINE, HTML_END_LINE)
+            ?.replace(TITLE_OPEN_NOTATION.toRegex(), TAG_B_OPEN)
+            ?.replace(TITLE_CLOSE_NOTATION.toRegex(), TAG_B_CLOSE)
+            ?.trim() ?: ""
```
Comment (thonc-2135) [inline app/src/main/java/com/edu/sun/oereminder/data/model/Message.kt]: vulue check null ở đầu r thì lm sao phải ? nhiều v.   replace lm sao mà null được nữa

**Review by tuanpa-0908 — APPROVED** [review]

**Pull request merged**